### PR TITLE
Added loading and disabled the button while searching for server

### PR DESCRIPTION
**Pull request by jaiwanth-v**
Fixes vector-im/element-web#16273 

ezgif com-gif-maker

Signed-off-by: Jaiwanth <jaiwanth2011@gmail.com>

### CODE REVIEW

File: src/components/views/dialogs/TextInputDialog.js [hunk 1]
```diff
@@ -51,6 +52,8 @@ export default class TextInputDialog extends React.Component {
 
         this.state = {
             value: this.props.value,
+            loading: false,
+            buttonValue: this.props.button,
```
Comment (t3chguy) [inline src/components/views/dialogs/TextInputDialog.js:56]: do we need this or can it be derived based on loading? e.g `this.state.loading ? _t("Loading...") : this.props.button` in render

File: src/components/views/dialogs/TextInputDialog.js [hunk 2]
```diff
@@ -125,7 +136,8 @@ export default class TextInputDialog extends React.Component {
                     </div>
                 </form>
                 <DialogButtons
-                    primaryButton={this.props.button}
+                    primaryButton={this.state.buttonValue}
+                    disabled = {this.state.loading}
```
Comment (t3chguy) [inline src/components/views/dialogs/TextInputDialog.js:140]:
```suggestion
                    disabled={this.state.loading}
```

**Review by t3chguy — CHANGES_REQUESTED** [review]
Looks good to me other than the small nit where we can derive data rather than storing two fields

File: src/components/views/dialogs/TextInputDialog.js (hunk 1 above)
Comment (jaiwanth-v) [reply src/components/views/dialogs/TextInputDialog.js:56]: yeah it's not necessary

File: src/components/views/dialogs/TextInputDialog.js [hunk 3]
```diff
@@ -51,6 +52,7 @@ export default class TextInputDialog extends React.Component {
 
         this.state = {
             value: this.props.value,
+            loading: false,
```
Comment (t3chguy) [inline src/components/views/dialogs/TextInputDialog.js:55]: Could we rename this to `busy`

File: src/components/views/dialogs/TextInputDialog.js [hunk 4]
```diff
@@ -125,7 +129,8 @@ export default class TextInputDialog extends React.Component {
                     </div>
                 </form>
                 <DialogButtons
-                    primaryButton={this.props.button}
+                    primaryButton={this.state.loading ? _t("Loading...") : this.props.button}
```
Comment (t3chguy) [inline src/components/views/dialogs/TextInputDialog.js:132]: Could we have `Loading...` come in as a prop which the caller specifies to be more generic?

**Review by t3chguy — CHANGES_REQUESTED** [review]
Sorry just 2 more tiny nits, I didn't realise this was in a Generic modal.

File: src/components/views/dialogs/TextInputDialog.js (hunk 3 above)
Comment (jaiwanth-v) [reply src/components/views/dialogs/TextInputDialog.js:55]: Okay, I'll rename that

File: src/components/views/dialogs/TextInputDialog.js (hunk 4 above)
Comment (jaiwanth-v) [reply src/components/views/dialogs/TextInputDialog.js:132]: Yes we can have that as an optional prop

File: src/components/views/dialogs/TextInputDialog.js [hunk 5]
```diff
@@ -18,6 +18,7 @@ import React, {createRef} from 'react';
 import PropTypes from 'prop-types';
 import * as sdk from '../../../index';
 import Field from "../elements/Field";
+import { _t } from '../../../languageHandler';
```
Comment (t3chguy) [inline src/components/views/dialogs/TextInputDialog.js:21]:
```suggestion
import { _t, _td } from '../../../languageHandler';
```

File: src/components/views/dialogs/TextInputDialog.js [hunk 6]
```diff
@@ -29,6 +30,7 @@ export default class TextInputDialog extends React.Component {
         value: PropTypes.string,
         placeholder: PropTypes.string,
         button: PropTypes.string,
+        busyMessage: PropTypes.string,
```
Comment (t3chguy) [inline src/components/views/dialogs/TextInputDialog.js:33]:
```suggestion
        busyMessage: PropTypes.string, // pass _td string
```

File: src/components/views/dialogs/TextInputDialog.js [hunk 7]
```diff
@@ -40,6 +42,7 @@ export default class TextInputDialog extends React.Component {
         title: "",
         value: "",
         description: "",
+        busyMessage: "Loading...",
```
Comment (t3chguy) [inline src/components/views/dialogs/TextInputDialog.js:45]:
```suggestion
        busyMessage: _td("Loading..."),
```

for the i18n script to work

**Review by t3chguy — APPROVED** [review]
Just accept my suggested changes and that'll be that

**Comment by t3chguy** [comment]
Thanks for your contribution!

**Pull request merged**